### PR TITLE
Release: add instruction to update doap.rdf file as part of release process

### LINKED ISSUE #9522: Add instructions on updating `doap.rdf` in the how-to-release guide
### Feature Request / Improvement

Thanks for updating it in https://github.com/apache/iceberg/pull/9507, but this should be done as part of the release. I think we should add it to the [instructions](https://github.com/apache/iceberg/blob/main/site/docs/how-to-release.md).

### Query engine

None

### PR DESCRIPTION
**Pull request by jbonofre**
This closes #9522

### CODE REVIEW

**Comment by jbonofre** [comment]
@Fokko @ajantha-bhat can you guys take a look ? Thanks !

File: site/docs/how-to-release.md [hunk 1]
```diff
@@ -308,6 +308,39 @@ Create a PR in the `iceberg` repo to make revapi run on the new release. For an
...
+    <release>
+      <Version>
+        <name>x.y.z</name>
+        <created>yyyy-mm-dd</created>
+        <revision>x.y.z</revision>
+      </Version>
+    </release>
+```
+
+NB: we set only the last available release in the `doap.rdf` file. You can eventually list different versions, but we need to add multiple release sections:
```
Comment (ajantha-bhat) [inline site/docs/how-to-release.md:325]: Maybe we don't have to confuse the readers by giving option, like arrow and other project, we can just keep the info about updating to the latest version.

File: site/docs/how-to-release.md [hunk 2]
```diff
@@ -308,6 +308,39 @@ Create a PR in the `iceberg` repo to make revapi run on the new release. For an
 - Create a PR in the `iceberg` repo to add the new version to the github issue template. For an example see [this PR](https://github.com/apache/iceberg/pull/6287).
 - Draft [a new release to update Github](https://github.com/apache/iceberg/releases/new) to show the latest release. A changelog can be generated automatically using Github.
 
+#### Update DOAP (ASF Project Description)
+
+- Create a PR to update the release version in `doap.rdf` file, in the `<release/>` section:
```
Comment (ajantha-bhat) [inline site/docs/how-to-release.md:313]: nit: maybe we can hyperlink to doap.rdf file here 
https://github.com/apache/iceberg/blob/main/doap.rdf

File: site/docs/how-to-release.md (hunk 1 above)
Comment (ajantha-bhat) [reply site/docs/how-to-release.md:325]: Above section we can mention that just update the version to latest release.

Comment (jbonofre) [reply site/docs/how-to-release.md:325]: OK, fair enough. Let me remove the Nota Bene 😄

**Review by ajantha-bhat — APPROVED** [review]

**Comment by jbonofre** [comment]
@Fokko what do you think about this PR ? Thanks !

**Comment by jbonofre** [comment]
@Fokko @nastra wdyt about this one guys ? Thanks !

**Pull request merged**